examples/OECC/README.md

```diff
@@ -0,0 +1,143 @@
+# OpenEdge Command Center #
+
+Demo scripts to setup an OpenEdge Command Center environment with support for OpenTelemetry metrics.
+
+The scripts can be run on a virtual machine running Ubuntu Linux 22.04.
+The VM could be created on AWS, Azure or other virtualization environment.
+Scripts run commands to detect the IP address of the machine.
+
+### Requirements 
+* Ubuntu Linux 22.04
+    * user with sudo access
+* OpenEdge media:
+    * PROGRESS_OE_12.8.3_LNX_64.tar.gz
+    * PROGRESS_OECC_SERVER_1.3.0_LNX_64.tar.gz
+    * PROGRESS_OECC_AGENT_1.3.0_LNX_64.bin
+* Reponse file for OpenEdge installation:
+    * response_12.8.3.ini
+    * Products:
+        * OE Adv. Ent. RDBMS
+        * 4GL Development System
+        * Progress ProdAS for OE
+
+**Notes:**
+* You can use the `OpenEdge-Samples/examples/OECC/config/response_12.8.3.ini` file as a template and specify the control codes in the file.
+* The demo scripts install OpenJDK to `/usr/lib/jvm/jdk` and OpenEdge to `/psc/dlc`.
+
+### Components 
+The infrastructure uses multiple components. To simplify the deployment, it uses a single virtual machine with some components running at the OS level and others running as a container using Docker.
+
+* Ubuntu Virtual Machine latest 22.04 (tested 22.04.4)
+    * OpenEdge Database 12.8.3
+    * PAS for OpenEdge 12.8.3
+    * OpenEdge Command Center agent 1.3.0
+    * Docker (latest, version tested: 26.1.4)
```

Comment:
How should a user install Docker for Ubuntu? Could we add a link to something like this page or embed some instructions? https://docs.docker.com/engine/install/ubuntu/

Reply:
Calling ./scripts/setup.sh would call ../common/scripts/install_docker.sh which performs the installation of Docker on Ubuntu.

Reply:
I was honestly not aware of that, and apparently did not look that closely at the setup.sh script to realize.

This is why I've been incredibly pedantic during the code review and this is exactly what has stopped me from proceeding further than I would have liked. There are a LOT of assumptions made about the skillset of the user, familiarity with the technology, and perhaps some details missed because you are writing as the subject matter expert. It's easy to miss things or take for granted what you already know. I would ask that you take another look at the opening of the main README file, specifically the Requirements and Components sections to add notes about where each should come from.

For example, for the OpenEdge media, state those should come from the ESD site, and where they might be found. For the components such as Docker, Grafana, etc. state that those are installed by the scripts if that's the case--otherwise, explain how to manually install them (apt-get, etc.) so users know they need to have them installed before proceeding.

Reply:
As someone who has had little exposure to any of this, I'm now curious how much is OECC and how much is for OTel? Would this have been better to demonstrate as 2 separate examples? OECC should just be OECC, OTel just OTel. By mixing components it's difficult to know which is required or optional for their respective technology.

We already have a separate Kafka folder that demonstrates just that technology--why mix these concepts under OECC?

Reply:
If feel that we can go either way on this.

The idea of having both together was because, at the time that the initial scripts were written and perhaps still today, OpenTelemetry is one of the most important features when using OpenEdge Command Center.
Related link: https://www.progress.com/openedge/features/command-center

It might be likely that users exploring OpenEdge Command Center (server) would also be interested on taking a look at the OpenTelemetry support.

A demo for OpenTelemetry only needs the configuration of OpenEdge Command Center agent.
Perhaps, we can have a simplified version of the demo scripts that only needs OpenEdge Command Center agent.
The interface to setup the demo would still be the same:
~~~
cd <demo-directory>
./scripts/setup.sh
~~~

Thanks.

Reply:
Where are you expecting these subtopics?
Are thinking on the being part of OpenEdge-Samples/examples/OECC?
Or are you think on separate demos?

The current README.md file has a "Creating a Dashboard in Grafana" as a separate section on the demo.

I can add an optional section to the README.md on "Using OpenTelemetry Support in OpenEdge" and leave the core of the README.md to be OpenEdge Command Center.

Please confirm.

Thanks.

Reply:
Where did this work originate? Do we have a ticket that explains the purpose of the work, acceptance criteria, etc? Or is this all just off-the-cuff? We should really be tracking the work and should have discussed what goes into these examples as a matter of agreeing on the content. The more I'm diving into this the more confused I am over why I would need Docker and all these other tools just to install something that already has installer files. The OECC itself would be on a central server/location while the agent would be installed on multiple client machines, so it's hard to visualize using this in a distributed manner.

If the need for Docker is so the instance can be easily encapsulated, then there should be a Docker container for the OECC (server) component while a separate container with a PASOE instance and the agent would be installed together. That way it could demonstrate how to set up each side of the solution. This could all be a single document that is the OECC example itself--nothing more. The addition of OTel, Grafana, etc. just seem extraneous and should be it's own topic for after you've successfully set up OECC to monitor a PAS instance. That way the user could deploy the things needed to each side--either the server side of the components or the agent-monitored instances.